src/ngraph_conversions.h

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include "ngraph_builder.h"
+#include "ngraph_log.h"
+
+namespace ngraph_bridge {
+
+template<size_t a, size_t b, size_t c, size_t d>
+void Reshape(std::shared_ptr<ng::Node>& ng_node) {
```

Comment:
I think it's better to return a `shared_ptr` here than it is to mutate through the reference. There may be situations where we want to hang on to the original node as well.

I feel like this needs a more descriptive name. Maybe something like `NGraphDimShuffle`.

Would be cool if we could get this for dimension counts ranging from 0 (yup, really) up to, I dunno, about 6 or 7, along with a dynamic version (one that just takes a vector for the permutation and does the checks at runtime). No need to do it in this PR, though.

Reply:
Hmmm, so far every use case in the code just mutates the existing node.

Reply:
I'm ok with renaming it -- @avijit-nervana any thoughts? -- except i dislike `Shuffle` in the name as it kind of implies randomness to me.